hoomd/mpcd/ReverseNonequilibriumShearFlow.cc

```diff
@@ -0,0 +1,431 @@
+    ArrayHandle<Scalar2> h_particles_staged(m_particles_staged,
+                                            access_location::host,
+                                            access_mode::overwrite);
+    m_num_staged = 0;
+    unsigned int lo_idx = 0;
+    unsigned int hi_idx = 0;
+    for (unsigned int i = 0; i < num_pairs; ++i)
+        {
+#ifdef ENABLE_MPI
+        if (m_sysdef->isDomainDecomposed())
+            {
+            const auto mpi_comm = m_exec_conf->getMPICommunicator();
+            const int rank = m_exec_conf->getRank();
+
+            // find particle in lo slab closest to +p on all ranks
+            Scalar_Int lo_swap;
+            lo_swap.i = rank;
+            if (std::isinf(m_target_momentum))
+                {
+                lo_swap.s = (lo_idx < m_num_lo) ? h_particles_lo.data[lo_idx].y
+                                                : std::numeric_limits<Scalar>::min();
+                MPI_Allreduce(MPI_IN_PLACE,
+                              &lo_swap,
+                              1,
+                              MPI_HOOMD_SCALAR_INT,
+                              MPI_MAXLOC,
+                              mpi_comm);
+                }
+            else
+                {
+                lo_swap.s = (lo_idx < m_num_lo)
+                                ? std::fabs(h_particles_lo.data[lo_idx].y - m_target_momentum)
+                                : std::numeric_limits<Scalar>::max();
+                MPI_Allreduce(MPI_IN_PLACE,
+                              &lo_swap,
+                              1,
+                              MPI_HOOMD_SCALAR_INT,
+                              MPI_MINLOC,
+                              mpi_comm);
+                }
+
+            // find particle in hi slab closest to -p on all ranks
+            Scalar_Int hi_swap;
+            hi_swap.i = rank;
+            if (std::isinf(m_target_momentum))
+                {
+                hi_swap.s = (hi_idx < m_num_hi) ? std::fabs(h_particles_hi.data[hi_idx].y)
+                                                : std::numeric_limits<Scalar>::min();
+                MPI_Allreduce(MPI_IN_PLACE,
+                              &hi_swap,
+                              1,
+                              MPI_HOOMD_SCALAR_INT,
+                              MPI_MAXLOC,
+                              mpi_comm);
+                }
+            else
+                {
+                hi_swap.s = (hi_idx < m_num_hi)
+                                ? std::fabs(h_particles_hi.data[hi_idx].y + m_target_momentum)
+                                : std::numeric_limits<Scalar>::max();
+                MPI_Allreduce(MPI_IN_PLACE,
+                              &hi_swap,
+                              1,
+                              MPI_HOOMD_SCALAR_INT,
+                              MPI_MINLOC,
+                              mpi_comm);
+                }
```

Comment:
Here, you should be able to always use MINLOC in conjunction with `max` if you don't use `fabs`.
```suggestion
            // find particle in hi slab to swap (p < 0)
            Scalar_Int hi_swap;
            hi_swap.i = rank;
            if (hi_idx < m_num_hi)
                {
                hi_swap.s = (std::isinf(m_target_momentum))
                        ? h_particles_hi.data[hi_idx].y
                        : std::fabs(h_particles_hi.data[hi_idx].y + m_target_momentum);
                }
            else
                {
                hi_swap.s = std::numeric_limits<Scalar>::max();
                }
                MPI_Allreduce(MPI_IN_PLACE,
                              &hi_swap,
                              1,
                              MPI_HOOMD_SCALAR_INT,
                              MPI_MINLOC,
                              mpi_comm);
```